### PR TITLE
add decision letters

### PR DESCRIPTION
**Pull request by aherzberg**
## Summary
Creates new mobile decision letters index endpoint

## Related issue(s)
https://github.com/department-of-veterans-affairs/va-mobile-app/issues/4896

## Testing done

- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected*
- *Describe the tests completed and the results*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: modules/mobile/spec/request/decision_letters_request_spec.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  before do
+    allow(VBMS::Client).to receive(:from_env_vars).and_return(FakeVBMS.new)
+    allow_any_instance_of(IAMUser).to receive(:icn).and_return('24811694708759028')
+    iam_sign_in(build(:iam_user))
+  end
+
+  # This endpoint's upstream service mocks it's own data for test env. HTTP client is not exposed by the
+  # connect_vbms gem so it cannot intercept the actual HTTP request, making the use of VCRs not possible.
+  # This means we cannot test error states for the index endpoint within specs
+  describe 'GET /mobile/v0/decision_letters' do
```
Comment (Tonksthebear) [inline modules/mobile/spec/request/decision_letters_request_spec.rb:20]: Is it worth adding tests for invalid responses?

Comment (aherzberg) [reply modules/mobile/spec/request/decision_letters_request_spec.rb:20]: unfortunately the upstream service doesn't have the ability to be used with VCRs so it just directly mocks the data. so the only response we can get out of it is valid data in test env. I played with manually created a faraday bad response but wasn't getting anywhere with it.

**Review by Tonksthebear — APPROVED** [review]
LGTM 🚀

File: modules/mobile/docs/schemas/DecisionLetterRecord.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      source:
+        type: string, null
+        example: "VBMS"
+      mimeType:
+        type: string, null
+        example: "application/pdf"
+      altDocTypes:
+        type: string, null
+        example: ""
+      restricted:
```
Comment (kpethtel) [inline modules/mobile/docs/schemas/DecisionLetterRecord.yml:59]: The model says restricted is not optional, so I don't think you need the null here.

File: modules/mobile/spec/support/schemas/decision_letter.json [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                  "null"
+                ]
+              },
+              "subject": {
+                "type": [
+                  "string",
+                  "null"
+                ]
+              },
+              "receivedAt": {
```
Comment (kpethtel) [inline modules/mobile/spec/support/schemas/decision_letter.json:85]: This also probably doesn't need the null option.

**Review by kpethtel — APPROVED** [review]
Couple of very small issues in the schemas but otherwise looks good.

File: modules/mobile/spec/support/schemas/decision_letter.json (hunk 3 above)
Comment (aherzberg) [reply modules/mobile/spec/support/schemas/decision_letter.json:85]: good catches. fixed

**Review by kpethtel — APPROVED** [review]

**Review by Tonksthebear — APPROVED** [review]
LGTM 🚀

**Pull request merged**